cmd/controller-manager/app/controller_manager.go

```diff
@@ -34,9 +35,13 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/tools/record"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apiserver/pkg/server/healthz"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
```

Comment:
> I would like to, but the EndpointsLock.Client is of type clientset.Interface as defined in the kube tree. We would need to move the leader election code into client-go for this to work and that would require changes in kube.

Work is in progress.  See https://github.com/kubernetes/client-go/issues/28, https://github.com/kubernetes/kubernetes/pull/39173, and https://github.com/kubernetes/kubernetes/issues/44338